### PR TITLE
MedPipe3D_JSOC_proposal

### PR DESCRIPTION
**Pull request by jakubMitura14**
Adding JSOC proposal.

### CODE REVIEW

**Review by jakubMitura14 — COMMENTED** [review]
> Adding JSOC proposal. Suggested reviewer Jacob Zelko 

@TheCedarPrince Cedar prince do not display on ossible reviewer list

File: jsoc/gsoc/medpipe3d.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# MedPipe3D Projects 
+
+MedPipe3D.jl together with MedEye3D.jl MedEval3D.jl and currently in development MedImage.jl is a set of libraries created to provide essential tools for 3D medical imaging to the Julia language ecosystem. Recently all of the projects migrated also to the JuliaHealth organization to improve packages visibility and promote interoperability.
```
Comment (TheCedarPrince) [inline jsoc/gsoc/medpipe3d.md:3]: Some comments on this piece: 

1. Could you link the repositories here for each package you mentioned?
2. Could you make a markdown list to explain what (in just a few words) what each package does generally?
3. I'd remove the line about package migration.

File: jsoc/gsoc/medpipe3d.md
Comment (TheCedarPrince) [inline jsoc/gsoc/medpipe3d.md:1]: Could you actually make the PR against this file: https://github.com/JuliaLang/www.julialang.org/blob/c35d1b42090ce631a46a221996446d3ecc9c6700/jsoc/gsoc/juliahealth.md?plain=1 ?

And then add a section for the Medical Imaging Subecosystem with the projects you have here listed underneath it?

File: jsoc/gsoc/medpipe3d.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- **Mentor:** Jakub Mitura [email: jakub.mitura14@gmail.com]
+
+- **Difficulty**: Hard
+
+- **Duration**: 350 hours
+
+- **Suggested Skills and Background**: 
+  - Experience with Julia
+  - Basic familiarity with computer graphics preferably OpenGL 
+  - Some experience with 3d volumetric data with spatial metadata (or a willingness to learn!)
```
Comment (TheCedarPrince) [inline jsoc/gsoc/medpipe3d.md:22]: Just a suggestion, but in the past, students have found it helpful when I list what packages to be familiar with as well as any links to requisite background knowledge (youtube videos, articles, etc.). Might want to consider adding them in.

File: jsoc/gsoc/medpipe3d.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+4) Any measurable decrease in the start time of the viewer -   20h
+5) The user will be able to display images just by supplying MedImage objects from the MedImage.jl library to a single display function -  20h
+
+
+
+
+### Project 2: Adding dataset-wide functions and integrations of augmentations
+
+**Description:** 
+MedPipe3D was created as a package that improves integration between other parts of the small ecosystem (MedEye3D, MedEval3D, and MedImage). Currently, it needs to be expanded and adapted so it can be a basis for a fully functional medical imaging pipeline.
```
Comment (TheCedarPrince) [inline jsoc/gsoc/medpipe3d.md:46]: Could you expand this description further to say in what ways it needs to be expanded/adapted?

File: jsoc/gsoc/medpipe3d.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+5) Add basic post-processing like the largest connected component analysis.
+6) Set all hyperparameters (of augmentation; size of a patch, threshold for getting binary mask from probabilities) in a struct or dictionary to enable hyperparameter tuning.
+7) Enable automated display of the algorithm output in the validation epoch, including saving such outputs to persistent storage.
+8) Support k-fold cross-validation.
+
+This set of changes although time-consuming to implement should not pose a significant issue to anybody with experience with the Julia programming language. However, implementing those will be a huge step in making Julia language a good alternative to Python in developing end-to-end medical imaging segmentation algorithms.
+
+- **Success criteria and time needed:** How the success of functionality described above is defined and the approximate time required for each.
+For each point mentor will also supply the person responsible for implementation with examples of required functionalities in Python or will point to the Julia libraries already implementing it (that just need to be integrated).
```
Comment (TheCedarPrince) [inline jsoc/gsoc/medpipe3d.md:74]: Remove line?

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
This looks fantastic! Left some comments/suggestions. The biggest comment is just the first comment -- feel free to integrate or reject the others as you see fit. Exciting! :smile:

**Review by TheCedarPrince — APPROVED** [review]

**Pull request merged**